### ISSUE TITLE
[TT-3158] path and match_pattern does not behave the same with regular expression

### ISSUE THREAD

**Issue by ghost** [TykTechnologies/tyk#3547]
**Branch/Environment/Version**
- Branch/Version: 3.0.5
- Environment: On-prem

**Describe the bug**
`path` and `match_pattern` does not behave the same causing problems on `url_rewrites`, `method_transforms` and `transform_headers` in `extended_paths`

**Reproduction steps**
Bellow is an API configuration that helps to reproduce the steps.

1. Load the API
2. Create a GET request on `http://127.0.0.1:8080/base/v1/resource/abcd12345`
3. The request is proxied correctly to the target_url, but the `method_transforms` and `transform_headers` rules are applied to it, even if the regular expresion is not matching (it seems the the lenght param in the regex is ignored). The regex `(?:[0-9a-fA-F]{8})$` is treated like `(?:[0-9a-fA-F]([^/]*))$`

**Actual behavior**

```
GET http://127.0.0.1:8080/base/v1/resource/abcd12345
x-remove-header: remove-value
x-test-header: test-value
```
The above request has the following results:
* Method GET will be transformed to POST
* header `x-remove-header` is removed
* header `x-add-header` is added
* POST request is proxied to target_url: https://webhook.site/0c3d0806-3840-4c11-8677-3dd54307d641

**Expected behavior**

```
GET http://127.0.0.1:8080/base/v1/resource/abcd12345
x-remove-header: remove-value
x-test-header: test-value
```
The above request should have the following results:
* Method GET remains
* header `x-remove-header` is not removed
* header `x-add-header` is not added
* GET request is proxied to target_url: https://webhook.site/0c3d0806-3840-4c11-8677-3dd54307d641

**Working examples**
##### Case that mathes regex

```
GET http://127.0.0.1:8080/base/v1/resource/abcd1234
x-remove-header: value
x-test-header: value
```

The above request has the following results:
* Method GET will be transformed to POST
* header `x-remove-header` is removed
* header `x-add-header` is added
* POST request is proxied to https://httpbin.org/post?id=abcd1234

##### Case that does not math regex (by content, not by length)

```
GET http://127.0.0.1:8080/base/v1/resource/test1234
x-remove-header: remove-value
x-test-header: test-value
```
The above request has the following results:
* Method GET remains
* header `x-remove-header` is not removed
* header `x-add-header` is not added
* GET request is proxied to target_url: https://webhook.site/0c3d0806-3840-4c11-8677-3dd54307d641

**Configuration (tyk api config file):**
```json
{
  "name": "API - v1",
  "slug": "base_v1",
  "listen_port": 0,
  "protocol": "http",
  "enable_proxy_protocol": false,
  "api_id": "base_v1",
  "org_id": "",
  "use_keyless": true,
  "use_oauth2": false,
  "use_openid": false,
  "openid_options": {
    "providers": null,
    "segregate_by_client": false
  },
  "oauth_meta": {
    "allowed_access_types": null,
    "allowed_authorize_types": null,
    "auth_login_redirect": ""
  },
  "auth": {
    "use_param": false,
    "param_name": "",
    "use_cookie": false,
    "cookie_name": "",
    "auth_header_name": "",
    "use_certificate": false,
    "validate_signature": false,
    "signature": {
      "algorithm": "",
      "header": "",
      "secret": "",
      "allowed_clock_skew": 0,
      "error_code": 0,
      "error_message": ""
    }
  },
  "auth_configs": null,
  "use_basic_auth": false,
  "basic_auth": {
    "disable_caching": false,
    "cache_ttl": 0,
    "extract_from_body": false,
    "body_user_regexp": "",
    "body_password_regexp": ""
  },
  "use_mutual_tls_auth": false,
  "client_certificates": null,
  "upstream_certificates": null,
  "pinned_public_keys": null,
  "enable_jwt": false,
  "use_standard_auth": false,
  "use_go_plugin_auth": false,
  "enable_coprocess_auth": false,
  "jwt_signing_method": "",
  "jwt_source": "",
  "jwt_identity_base_field": "",
  "jwt_client_base_field": "",
  "jwt_policy_field_name": "",
  "jwt_default_policies": null,
  "jwt_issued_at_validation_skew": 0,
  "jwt_expires_at_validation_skew": 0,
  "jwt_not_before_validation_skew": 0,
  "jwt_skip_kid": false,
  "jwt_scope_to_policy_mapping": null,
  "jwt_scope_claim_name": "",
  "notifications": {
    "shared_secret": "",
    "oauth_on_keychange_url": ""
  },
  "enable_signature_checking": false,
  "hmac_allowed_clock_skew": 0,
  "hmac_allowed_algorithms": null,
  "request_signing": {
    "is_enabled": false,
    "secret": "",
    "key_id": "",
    "algorithm": "",
    "header_list": null,
    "certificate_id": "",
    "signature_header": ""
  },
  "base_identity_provided_by": "",
  "definition": {
    "location": "url",
    "key": "",
    "strip_path": false
  },
  "version_data": {
    "not_versioned": true,
    "default_version": "",
    "versions": {
      "Default": {
        "name": "Default",
        "expires": "",
        "paths": {
          "ignored": null,
          "white_list": null,
          "black_list": null
        },
        "use_extended_paths": true,
        "extended_paths": {
          "transform_headers": [
            {
              "delete_headers": [
                "x-remove-header"
              ],
              "add_headers": {
                "x-add-header": "add-value"
              },
              "path": "/resource/((?:[0-9a-fA-F]{8}))$",
              "method": "GET",
              "act_on": false
            }
          ],
          "url_rewrites": [
            {
              "path": "/resource/((?:[0-9a-fA-F]{8}))$",
              "method": "GET",
              "match_pattern": "^/base/v1/resource/((?:[0-9a-fA-F]{8}))$",
              "rewrite_to": "https://httpbin.org/post?id=$1",
              "triggers": null
            }
          ],
          "method_transforms": [
            {
              "path": "/resource/((?:[0-9a-fA-F]{8}))$",
              "method": "GET",
              "to_method": "POST"
            }
          ]
        },
        "global_headers": null,
        "global_headers_remove": null,
        "global_response_headers": null,
        "global_response_headers_remove": null,
        "ignore_endpoint_case": false,
        "global_size_limit": 0,
        "override_target": ""
      }
    }
  },
  "uptime_tests": {
    "check_list": null,
    "config": {
      "expire_utime_after": 0,
      "service_discovery": {
        "use_discovery_service": false,
        "query_endpoint": "",
        "use_nested_query": false,
        "parent_data_path": "",
        "data_path": "",
        "port_data_path": "",
        "target_path": "",
        "use_target_list": false,
        "cache_timeout": 0,
        "endpoint_returns_list": false
      },
      "recheck_wait": 0
    }
  },
  "proxy": {
    "preserve_host_header": false,
    "listen_path": "/base/v1/",
    "target_url": "https://webhook.site/0c3d0806-3840-4c11-8677-3dd54307d641",
    "disable_strip_slash": false,
    "strip_listen_path": false,
    "enable_load_balancing": false,
    "target_list": null,
    "check_host_against_uptime_tests": false,
    "service_discovery": {
      "use_discovery_service": false,
      "query_endpoint": "",
      "use_nested_query": false,
      "parent_data_path": "",
      "data_path": "",
      "port_data_path": "",
      "target_path": "",
      "use_target_list": false,
      "cache_timeout": 0,
      "endpoint_returns_list": false
    },
    "transport": {
      "ssl_insecure_skip_verify": false,
      "ssl_ciphers": null,
      "ssl_min_version": 0,
      "ssl_force_common_name_check": false,
      "proxy_url": ""
    }
  },
  "disable_rate_limit": false,
  "disable_quota": false,
  "custom_middleware": {
    "pre": [],
    "post": null,
    "post_key_auth": null,
    "auth_check": {
      "name": "",
      "path": "",
      "require_session": false,
      "raw_body_only": false
    },
    "response": null,
    "driver": "",
    "id_extractor": {
      "extract_from": "",
      "extract_with": "",
      "extractor_config": null
    }
  },
  "custom_middleware_bundle": "",
  "cache_options": {
    "cache_timeout": 0,
    "enable_cache": false,
    "cache_all_safe_requests": false,
    "cache_response_codes": null,
    "enable_upstream_cache_control": false,
    "cache_control_ttl_header": "",
    "cache_by_headers": null
  },
  "session_lifetime": 0,
  "active": false,
  "internal": false,
  "auth_provider": {
    "name": "",
    "storage_engine": "",
    "meta": null
  },
  "session_provider": {
    "name": "",
    "storage_engine": "",
    "meta": null
  },
  "event_handlers": {
    "events": null
  },
  "enable_batch_request_support": false,
  "enable_ip_whitelisting": false,
  "allowed_ips": null,
  "enable_ip_blacklisting": false,
  "blacklisted_ips": null,
  "dont_set_quota_on_create": false,
  "expire_analytics_after": 0,
  "response_processors": null,
  "CORS": {
    "enable": false,
    "allowed_origins": null,
    "allowed_methods": null,
    "allowed_headers": null,
    "exposed_headers": null,
    "allow_credentials": false,
    "max_age": 0,
    "options_passthrough": false,
    "debug": false
  },
  "domain": "",
  "certificates": null,
  "do_not_track": false,
  "tags": [],
  "enable_context_vars": false,
  "config_data": {},
  "tag_headers": null,
  "global_rate_limit": {
    "rate": 0,
    "per": 0
  },
  "strip_auth_data": false,
  "enable_detailed_recording": false,
  "graphql": {
    "enabled": false,
    "execution_mode": "",
    "schema": "",
    "type_field_configurations": null,
    "playground": {
      "enabled": false,
      "path": ""
    }
  }
}
```

**Comment by Moses-oyedeji** [comment]
@anx-hnezbeda  Thanks for contacting Tyk!
We apologize for the delayed response. The issue is currently being reviewed and we will keep you updated as soon as we can

**Comment by andyo-tyk** [comment]
This was fixed in Tyk 5.5.1.
